code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -22,93 +23,123 @@
+	var/robot_arm = /obj/item/bodypart/r_arm/robot
+	///People currently looking into a bot's UI panel.
+	var/list/users = list()
+	///The inserted (if any) pAI in this bot.
+	var/obj/item/paicard/paicard
+	///The type of bot it is, for radio control.
+	var/bot_type = NONE
+
+	///Additonal access given to player-controlled bots.
+	var/list/player_access = list()
+	///All initial access this bot started with.
 	var/list/prev_access = list()
-	var/on = TRUE
-	var/open = FALSE//Maint panel
-	var/locked = TRUE
-	///If the bot is hacked by silicons or emagged by humans.
-	var/hacked = FALSE
-	var/text_hack = "" //Custom text returned to a silicon upon hacking a bot.
-	var/text_dehack = "" //Text shown when resetting a bots hacked status to normal.
-	var/text_dehack_fail = "" //Shown when a silicon tries to reset a bot emagged with the emag item, which cannot be reset.
-	var/declare_message = "" //What the bot will display to the HUD user.
-	var/frustration = 0 //Used by some bots for tracking failures to reach their target.
-	var/base_speed = 2 //The speed at which the bot moves, or the number of times it moves per process() tick.
-	var/turf/ai_waypoint //The end point of a bot's path, or the target location.
-	var/list/path = list() //List of turfs through which a bot 'steps' to reach the waypoint, associated with the path image, if there is one.
-	var/pathset = 0
-	var/list/ignore_list = list() //List of unreachable targets for an ignore-list enabled bot to ignore.
-	var/mode = BOT_IDLE //Standardizes the vars that indicate the bot is busy with its function.
-	var/tries = 0 //Number of times the bot tried and failed to move.
-	var/remote_disabled = FALSE //If enabled, the AI cannot *Remotely* control a bot. It can still control it through cameras.
-	var/mob/living/silicon/ai/calling_ai //Links a bot to the AI calling it.
-	var/obj/item/radio/Radio //The bot's radio, for speaking to people.
-	var/radio_key = null //which channels can the bot listen to
-	var/radio_channel = RADIO_CHANNEL_COMMON //The bot's default radio channel
-	var/auto_patrol = 0// set to make bot automatically patrol
-	var/turf/patrol_target // this is turf to navigate to (location of beacon)
-	var/turf/summon_target // The turf of a user summoning a bot.
-	var/new_destination // pending new destination (waiting for beacon response)
-	var/destination // destination description tag
-	var/next_destination // the next destination in the patrol route
-	var/shuffle = FALSE // If we should shuffle our adjacency checking
-
-	var/blockcount = 0 //number of times retried a blocked path
-	var/awaiting_beacon = 0 // count of pticks awaiting a beacon response
-
-	var/nearest_beacon // the nearest beacon's tag
-	var/turf/nearest_beacon_loc // the nearest beacon's location
+
+	///Bot-related mode flags on the Bot indicating how they will act.
+	var/bot_mode_flags = BOT_MODE_ON | BOT_MODE_REMOTE_ENABLED | BOT_MODE_PAI_CONTROLLABLE
```

Comment:
why are you moving so many of these important vars into bitflags?

Reply:
I thought it was better to have 1 flag than 5 different vars for things that all have some common theme.